### PR TITLE
Bubble distance check

### PR DESCRIPTION
**Pull request by AisenGinn**
Added distance check between vehicle and bubbles to avoid unnacessary creation of bubble.

### CODE REVIEW

File: smarts/core/bubble_manager.py [hunk 1]
```diff
@@ -538,17 +550,31 @@ def _sync_cursors(self, last_vehicle_index, vehicle_index):
...
-                    index=persisted_vehicle_index,
-                    vehicle_ids_per_bubble=vehicle_ids_per_bubble,
-                    running_cursors=cursors,
+            point = vehicle.pose.point
+            v_radius = (
+                math.sqrt(
+                    vehicle.width * vehicle.width + vehicle.length * vehicle.length
                 )
-                cursors.add(cursor)
+            )
```
Comment (saulfield) [inline smarts/core/bubble_manager.py:558]: These outer parentheses seem to be unnecessary.

File: examples/single_agent.py [hunk 2]
```diff
@@ -70,6 +71,7 @@ def main(scenarios, headless, num_episodes, max_episode_steps=None):
 if __name__ == "__main__":
     parser = default_argument_parser("single-agent-example")
     args = parser.parse_args()
+    sys.path.append(".")
```
Comment (saulfield) [inline examples/single_agent.py:74]: Why was this needed?

Comment (AisenGinn) [reply examples/single_agent.py:74]: I cannot run the simulation sometimes without this line, I will remove this once other changes are approved

**Review by saulfield — APPROVED** [review]

File: smarts/core/bubble_manager.py [hunk 3]
```diff
@@ -76,6 +77,13 @@ class Bubble:
     def __init__(self, bubble: SSBubble, road_map: RoadMap):
         geometry = bubble.zone.to_geometry(road_map)
 
+        bounding = geometry.bounds
+        self.radius = math.sqrt(
+            (bounding[2] - bounding[0]) * (bounding[2] - bounding[0])
+            + (bounding[3] - bounding[1]) * (bounding[3] - bounding[1])
+        )
```
Comment (Gamenot) [inline smarts/core/bubble_manager.py:84]: This does not seem quite right... 

`pow(maxx - minxx, 2) + pow(maxy - miny, 2)`

I think this ends up significantly larger than it should be. I think it should be something like `pow(maxx - centroidx, 2) + pow(centroidy - miny, 2) + 0.1` perhaps?

**Review by Gamenot — CHANGES_REQUESTED** [review]
This ended up more compact than I expected.

**Review by Gamenot — APPROVED** [review]
